src/nvim/buffer.c

```diff
@@ -1458,7 +1460,9 @@ buflist_new (
     }
     lastbuf = buf;
 
-    buf->b_fnum = top_file_num++;
+    // If bufnr is nonzero it is assumed to be a previously
+    // reserved buffer number (handle)
+    buf->handle = bufnr != 0 ? bufnr : top_file_num++;
```

Comment:
They are synonymous, but probably better `b_fnum` yes (I reckon we won't change `b_fnum` to `handle` generally, to avoid friction merging vim patches)